transport/peer/x/peerlist/roundrobin/roundrobin.go

```diff
@@ -67,6 +70,24 @@ func (pl *RoundRobin) addAll(peerIDs []transport.PeerIdentifier) error {
 	return errs
 }
 
+// Add a peer identifier to the round robin
+func (pl *RoundRobin) Add(pid transport.PeerIdentifier) error {
+	p, err := pl.agent.RetainPeer(pid, pl)
+	if err != nil {
+		return err
+	}
+
+	defer pl.notifyPeerAddedEvent()
```

Comment:
Avoid defer if there's a single exit path.

```
err := pl.pr.Add(p)
pl.notifyPeerAddedEvent()
return err
```

Also, do you really want to notify even if it failed to add?